src/screens/MainScreen/index.js

```diff
@@ -151,10 +159,19 @@ const MainScreen = () => {
         />
       )}
       {showMatch && (
-        <MatchModal match={createdTarget.matchedUser} onHide={() => setShowMatch(false)} />
+        <MatchModal
+          matchedUser={createdTarget.matchedUser}
+          matchConversation={createdTarget.matchConversation}
+          onHide={() => setShowMatch(false)}
+          onStartChatting={onStartChatting}
```

Comment:
You already know which ones are the user and the conversation from here so you can pass this
```suggestion
         onStartChatting ={() => onStartChatting({ matchedUser, matchConversation })}
```
and then just use `onPress={onStartChatting}` into MatchModal component